streaming/base/batching/per_stream.py

```diff
@@ -63,6 +63,7 @@ def generate_work_per_stream_batching(dataset: StreamingDataset, world: World, e
             # same as the ratio of the stream's samples to overall samples.
             # This ensures that the overall training shuffle block size is still approximately
             # equal to what is set by the user, and allows for reasoning about cache_limit as well.
+            assert isinstance(dataset.shuffle_block_size, int)
```

Comment:
Agreed, I'll change this one and some other ones where it makes sense to throw an exception! But many of the other assert statements were for linting purposes because pyright was throwing errors for cases that would never happen.

Reply:
Also, it's technically not possible for `dataset.shuffle_block_size` to not be an int in this case, it just happens in the `_resume` method so pyright doesn't recognize that. So this was also included because of linting and I don't really have a way to test that the exception is raised...